src/plugins/intel_gpu/src/graph/program.cpp

```diff
@@ -763,8 +763,7 @@ void program::cleanup() {
         }
     }
 
-    if (_engine.configuration().kernels_cache_path.empty())
-        _kernels_cache->reset();
+    _kernels_cache->reset();
```

Comment:
https://github.com/openvinotoolkit/openvino/pull/13801#discussion_r1021120002
Reverted. Thank you.